fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2912,6 +2950,7 @@ public void removeFormerIndex(FormerIndex formerIndex) {
         tr.clear(getSubspace().range(Tuple.from(INDEX_SECONDARY_SPACE_KEY, formerIndex.getSubspaceTupleKey())));
         tr.clear(getSubspace().range(Tuple.from(INDEX_RANGE_SPACE_KEY, formerIndex.getSubspaceTupleKey())));
         tr.clear(getSubspace().pack(Tuple.from(INDEX_STATE_SPACE_KEY, formerIndex.getSubspaceTupleKey())));
+        tr.clear(getSubspace().pack(Tuple.from(INDEX_STATE_SPACE_KEY_DEPRECATED, formerIndex.getFormerName())));
```

Comment:
In a `FormerIndex`, `getFormerName` can be `null` if the former name is dropped when the `FormerIndex` is created. (Which `RecordMetaDataBuilder::removeIndex` currently adds in the former name, but it used to not (prior to eef2a53).) I think this should guard the removal on whether `getFormerName` is `null` or not.

Reply:
sure. fixed